File: src/services/tts-service.js
import axios from 'axios';
import { config } from '../config/config.js';
import logger from '../config/logger.js';

class TTSService {
  constructor() {
    this.baseUrl = config.coquiTts.url;
  }

  async synthesize(text) {
    try {
      logger.info('Synthesizing speech', { text: text.substring(0, 50) });

      // Call Coqui TTS API with Thorsten voice
      const response = await axios.post(
        `${this.baseUrl}/api/tts`,
        {
          text: text,
          speaker_id: 'thorsten', // Thorsten voice
          language_id: 'de'
        },
        {
          responseType: 'arraybuffer',
          timeout: 30000
        }
      );

      logger.info('Speech synthesized successfully', { 
        size: response.data.byteLength 
      });

      return Buffer.from(response.data);
    } catch (error) {
      logger.error('Error synthesizing speech', { 
        error: error.message,
        text: text.substring(0, 50)
      });
      
      // Fallback: return empty buffer
      return Buffer.alloc(0);
    }
  }

  async getVoices() {
    try {
      const response = await axios.get(`${this.baseUrl}/api/voices`);
      return response.data;
    } catch (error) {
      logger.error('Error getting voices', { error: error.message });
      return [];
    }
  }
}

export default TTSService;

File: python/tts_service.py
"""
Text-to-Speech service using Coqui TTS
"""
import logging
import requests
from config import COQUI_TTS_URL

logger = logging.getLogger(__name__)


class TTSService:
    """Handles text-to-speech conversion"""
    
    def __init__(self):
        self.base_url = COQUI_TTS_URL
    
    async def synthesize(self, text: str) -> bytes:
        """
        Convert text to speech audio
        
        Args:
            text: Text to convert to speech
            
        Returns:
            Audio data as bytes
        """
        try:
            logger.info(f"Synthesizing: {text[:50]}...")
            
            response = requests.post(
                f"{self.base_url}/api/tts",
                json={
                    'text': text,
                    'speaker_id': 'thorsten',
                    'language_id': 'de'
                },
                timeout=30
            )
            
            response.raise_for_status()
            
            logger.info(f"Speech synthesized: {len(response.content)} bytes")
            return response.content
            
        except Exception as e:
            logger.error(f"Error synthesizing speech: {e}")
            # Return empty bytes on error
            return b''
